### ISSUE TITLE
removing placeholder text from media-only posts breaks compatibility with gnu social

### ISSUE THREAD

**Issue by rainyday** [tootsuite/mastodon#7856]
This PR: https://github.com/tootsuite/mastodon/pull/7806 breaks compatibility with GNU Social, Postactiv and potentially other ostatus servers.

Please consider reverting this as it affects more than just legacy Mastodon servers.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by Gargron** [comment]
Only text-free posts are incompatible, and people have been asking for it. Of course, this can be fixed manually in a given post, too, by entering any text or dot or an empty space character. Since the number of OStatus servers is going down and the protocol has been superseded by ActivityPub, I don't see this as a high priority issue, sorry.

**Comment by Gargron** [comment]
Did you mean to say breaking compatibility with [8%](https://fediverse.network/) of the servers out there?

**Comment by rainyday** [comment]
No, I don't, but even 8% should be enough to avoid breaking federation for a cosmetic issue.

**Comment by trwnh** [comment]
I'm not sure I understand that last point -- at what point does it suddenly become acceptable/unacceptable to break support with outdated and deprecated applications? To take an example from CSS and browsers: 96% of active browsers support Flexbox, and 85% support Grid. Does that mean we shouldn't write websites in Grid, or should we instead expect people to upgrade? For how long should one continue to support Internet Explorer, rather than simply accepting that some legacy users will have a somewhat-degraded experience? Of course, it still bears examination whether that degradation is graceful or completely breaking. If it is indeed merely a cosmetic issue, then by all means, break compatibility.

**Comment by rainyday** [comment]
@trwnh You misunderstand the issue. keeping compatibility is a a cosmetic issue *for mastodon*. breaking compatibility (by making some posts completely undeliverable) to solve a fix a minor cosmetic issue for mastodon is what I am arguing against.

I do agree with you that we should expect people to upgrade at some point however that isn't what's going on here. People on GNU Social (and other ostatus servers) do not have an option to upgrade since the latest versions of these servers are the ones with this issue. In this context the only "upgrade" they could do is to switch to a completely different server software altogether, losing all data in the process.

GNU Social can (with some justification) be called "legacy" software but it is still maintained. As far as I can tell, instead of reaching out to them and attempting to coordinate this change, @Gargron rolled out this "feature" without warning (or even an announcement) and expects everyone to discover it on their own and scramble to fix things after the fact which is incredibly bad form. And again, this is all just to fix a perceived minor cosmetic issue on the mastodon side.
